### PR TITLE
Fix NumPy 2.0 deprecations via running ruff

### LINKED ISSUE #2498: FloatCanvas triggers DeprecationWarning from Numpy 1.25.0 or later for 'sometrue'
**Operating system**: Linux Mint 21.2
**wxPython version & source**:  4.2.1  from pypi         
**Python version & source**: 3.10.12 from distro          
**Numpy version & source**: 1.26.2 from pypi

**Description of the problem**:

I copied the code from the [docs for the FloatCanvas package](https://docs.wxpython.org/wx.lib.floatcanvas.FloatCanvas.html#module-wx.lib.floatcanvas.FloatCanvas) into a small test app (see example below). 

When I ran the app, it output the following warning:
```
/home/richardt/Development/python/exp/hci/wxpython/float_canvas/float_canvas_1.py:25: DeprecationWarning: `sometrue` is deprecated as of NumPy 1.25.0, and will be removed in NumPy 2.0. Please use `any` instead.
  self.Canvas.Draw()
```

Line 604 in FloatCanvas.py for the FloatCanvas.Draw() method contains:
```
        if N.sometrue(self.PanelSize <= 2 ):
```

I did a recursive search of my wxPython 4.2.1 installation and found that `sometrue` also occurs in line 2047 of  `lib/plot/plotcanvas.py`:
```
             if np.sometrue(
```

<details>
  <summary>Code Example (click to expand)</summary>

```python
import wx
from wx.lib.floatcanvas import FloatCanvas

class MyFrame(wx.Frame):
    def __init__(self, parent):
        wx.Frame.__init__(self, parent)

        self.Canvas = FloatCanvas.FloatCanvas(self, -1,
                                     size=(500, 500),
                                     ProjectionFun=None,
                                     Debug=0,
                                     BackgroundColor="White",
                                     )

        # add a circle
        cir = FloatCanvas.Circle((10, 10), 100)
        self.Canvas.AddObject(cir)

        # add a rectangle
        rect = FloatCanvas.Rectangle((110, 10), (100, 100), FillColor='Red')
        self.Canvas.AddObject(rect)

        self.Canvas.Draw()

app = wx.App()
frame = MyFrame(None)
frame.Show()
app.MainLoop()

```
</details>

### LINKED ISSUE #2574: Numpy2 removed numpy.sometrue.
Windows 10 22H2
'4.2.1 msw (phoenix) wxWidgets 3.2.2.1'
Python 3.10.7 (tags/v3.10.7:6cc6b13, Sep  5 2022, 14:08:36) [MSC v.1933 64 bit (AMD64)] on win32

I am not yet sure exactly the code path that is calling this but I will track down how to trigger this in a simple way but my application generates an exception. The core issue is in floatcanvas/FloatCanvas.py at line 604 

        if N.sometrue(self.PanelSize <= 2 ):

numpy no longer has a numpy.sometrue() function

### PR DESCRIPTION
**Pull request by radoering**
I just run the following command:

```
ruff check --select NPY201 --fix --exclude docs/sphinx/rest_substitutions/snippets/python/converted
```

to fix numpy deprecations that were removed and thereby result in an exception with numpy 2.0.

See https://numpy.org/devdocs/numpy_2_0_migration_guide.html#numpy-2-0-migration-guide

Fixes #2574
Fixes #2498
Closes #2578

### CODE REVIEW

File: wx/lib/floatcanvas/FloatCanvas.py [hunk 1]
```diff
@@ -601,7 +601,7 @@ def Draw(self, Force=False):
 
         """
 
-        if N.sometrue(self.PanelSize <= 2 ):
+        if any(self.PanelSize <= 2 ):
```
Comment (komoto48g) [inline wx/lib/floatcanvas/FloatCanvas.py:604]: The better would be `N.any`

File: wx/lib/plot/plotcanvas.py [hunk 2]
```diff
@@ -2044,7 +2044,7 @@ def UpdatePointLabel(self, mDataDict):
         """
         if self.last_PointLabel is not None:
             # compare pointXY
-            if np.sometrue(
+            if any(
```
Comment (komoto48g) [inline wx/lib/plot/plotcanvas.py:2047]: np.any?

File: wx/lib/floatcanvas/FloatCanvas.py (hunk 1 above)
Comment (radoering) [reply wx/lib/floatcanvas/FloatCanvas.py]: Thanks. It looks like I have put too much trust in ruff. Fixed it and created an issue for ruff: astral-sh/ruff#12416

**Comment by newville** [comment]
Please merge.

**Comment by bnavigator** [comment]
The unit tests have been missed by this PR:

```python
[ 1232s] =================================== FAILURES ===================================
[ 1232s] __________________________ testFromPoints.testCreate ___________________________
[ 1232s] [gw0] linux -- Python 3.10.14 /usr/bin/python3.10
[ 1232s] self = <unittests.test_lib_floatcanvas_bbox.testFromPoints testMethod=testCreate>
[ 1232s] 
[ 1232s]     def testCreate(self):
[ 1232s]         Pts = N.array( ((5,2),
[ 1232s]                 (3,4),
[ 1232s]                 (1,6),
[ 1232s] >               ), N.float_ )
[ 1232s] 
[ 1232s] unittests/test_lib_floatcanvas_bbox.py:344: 
[ 1232s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[ 1232s] 
[ 1232s] attr = 'float_'
[ 1232s] 
[ 1232s]     def __getattr__(attr):
[ 1232s]         # Warn for expired attributes
[ 1232s]         import warnings
[ 1232s]     
[ 1232s]         if attr == "linalg":
[ 1232s]             import numpy.linalg as linalg
[ 1232s]             return linalg
[ 1232s]         elif attr == "fft":
[ 1232s]             import numpy.fft as fft
[ 1232s]             return fft
[ 1232s]         elif attr == "dtypes":
[ 1232s]             import numpy.dtypes as dtypes
[ 1232s]             return dtypes
[ 1232s]         elif attr == "random":
[ 1232s]             import numpy.random as random
[ 1232s]             return random
[ 1232s]         elif attr == "polynomial":
[ 1232s]             import numpy.polynomial as polynomial
[ 1232s]             return polynomial
[ 1232s]         elif attr == "ma":
[ 1232s]             import numpy.ma as ma
[ 1232s]             return ma
[ 1232s]         elif attr == "ctypeslib":
[ 1232s]             import numpy.ctypeslib as ctypeslib
[ 1232s]             return ctypeslib
[ 1232s]         elif attr == "exceptions":
[ 1232s]             import numpy.exceptions as exceptions
[ 1232s]             return exceptions
[ 1232s]         elif attr == "testing":
[ 1232s]             import numpy.testing as testing
[ 1232s]             return testing
[ 1232s]         elif attr == "matlib":
[ 1232s]             import numpy.matlib as matlib
[ 1232s]             return matlib
[ 1232s]         elif attr == "f2py":
[ 1232s]             import numpy.f2py as f2py
[ 1232s]             return f2py
[ 1232s]         elif attr == "typing":
[ 1232s]             import numpy.typing as typing
[ 1232s]             return typing
[ 1232s]         elif attr == "rec":
[ 1232s]             import numpy.rec as rec
[ 1232s]             return rec
[ 1232s]         elif attr == "char":
[ 1232s]             import numpy.char as char
[ 1232s]             return char
[ 1232s]         elif attr == "array_api":
[ 1232s]             raise AttributeError("`numpy.array_api` is not available from "
[ 1232s]                                  "numpy 2.0 onwards")
[ 1232s]         elif attr == "core":
[ 1232s]             import numpy.core as core
[ 1232s]             return core
[ 1232s]         elif attr == "strings":
[ 1232s]             import numpy.strings as strings
[ 1232s]             return strings
[ 1232s]         elif attr == "distutils":
[ 1232s]             if 'distutils' in __numpy_submodules__:
[ 1232s]                 import numpy.distutils as distutils
[ 1232s]                 return distutils
[ 1232s]             else:
[ 1232s]                 raise AttributeError("`numpy.distutils` is not available from "
[ 1232s]                                      "Python 3.12 onwards")
[ 1232s]     
[ 1232s]         if attr in __future_scalars__:
[ 1232s]             # And future warnings for those that will change, but also give
[ 1232s]             # the AttributeError
[ 1232s]             warnings.warn(
[ 1232s]                 f"In the future `np.{attr}` will be defined as the "
[ 1232s]                 "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
[ 1232s]     
[ 1232s]         if attr in __former_attrs__:
[ 1232s]             raise AttributeError(__former_attrs__[attr])
[ 1232s]     
[ 1232s]         if attr in __expired_attributes__:
[ 1232s] >           raise AttributeError(
[ 1232s]                 f"`np.{attr}` was removed in the NumPy 2.0 release. "
[ 1232s]                 f"{__expired_attributes__[attr]}"
[ 1232s]             )
[ 1232s] E           AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?
[ 1232s] 
[ 1232s] /usr/lib64/python3.10/site-packages/numpy/__init__.py:397: AttributeError
```

```
[ 1232s] ___________________________ testNullBBox.testValues ____________________________
[ 1232s] [gw0] linux -- Python 3.10.14 /usr/bin/python3.10
[ 1232s] self = <unittests.test_lib_floatcanvas_bbox.testNullBBox testMethod=testValues>
[ 1232s] 
[ 1232s]     def testValues(self):
[ 1232s] >       self.assertTrue( N.alltrue(N.isnan(self.B1)) )
[ 1232s] 
[ 1232s] unittests/test_lib_floatcanvas_bbox.py:463: 
[ 1232s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[ 1232s] 
[ 1232s] attr = 'alltrue'
[ 1232s] 
[ 1232s]     def __getattr__(attr):
[ 1232s]         # Warn for expired attributes
[ 1232s]         import warnings
[ 1232s]     
[ 1232s]         if attr == "linalg":
[ 1232s]             import numpy.linalg as linalg
[ 1232s]             return linalg
[ 1232s]         elif attr == "fft":
[ 1232s]             import numpy.fft as fft
[ 1232s]             return fft
[ 1232s]         elif attr == "dtypes":
[ 1232s]             import numpy.dtypes as dtypes
[ 1232s]             return dtypes
[ 1232s]         elif attr == "random":
[ 1232s]             import numpy.random as random
[ 1232s]             return random
[ 1232s]         elif attr == "polynomial":
[ 1232s]             import numpy.polynomial as polynomial
[ 1232s]             return polynomial
[ 1232s]         elif attr == "ma":
[ 1232s]             import numpy.ma as ma
[ 1232s]             return ma
[ 1232s]         elif attr == "ctypeslib":
[ 1232s]             import numpy.ctypeslib as ctypeslib
[ 1232s]             return ctypeslib
[ 1232s]         elif attr == "exceptions":
[ 1232s]             import numpy.exceptions as exceptions
[ 1232s]             return exceptions
[ 1232s]         elif attr == "testing":
[ 1232s]             import numpy.testing as testing
[ 1232s]             return testing
[ 1232s]         elif attr == "matlib":
[ 1232s]             import numpy.matlib as matlib
[ 1232s]             return matlib
[ 1232s]         elif attr == "f2py":
[ 1232s]             import numpy.f2py as f2py
[ 1232s]             return f2py
[ 1232s]         elif attr == "typing":
[ 1232s]             import numpy.typing as typing
[ 1232s]             return typing
[ 1232s]         elif attr == "rec":
[ 1232s]             import numpy.rec as rec
[ 1232s]             return rec
[ 1232s]         elif attr == "char":
[ 1232s]             import numpy.char as char
[ 1232s]             return char
[ 1232s]         elif attr == "array_api":
[ 1232s]             raise AttributeError("`numpy.array_api` is not available from "
[ 1232s]                                  "numpy 2.0 onwards")
[ 1232s]         elif attr == "core":
[ 1232s]             import numpy.core as core
[ 1232s]             return core
[ 1232s]         elif attr == "strings":
[ 1232s]             import numpy.strings as strings
[ 1232s]             return strings
[ 1232s]         elif attr == "distutils":
[ 1232s]             if 'distutils' in __numpy_submodules__:
[ 1232s]                 import numpy.distutils as distutils
[ 1232s]                 return distutils
[ 1232s]             else:
[ 1232s]                 raise AttributeError("`numpy.distutils` is not available from "
[ 1232s]                                      "Python 3.12 onwards")
[ 1232s]     
[ 1232s]         if attr in __future_scalars__:
[ 1232s]             # And future warnings for those that will change, but also give
[ 1232s]             # the AttributeError
[ 1232s]             warnings.warn(
[ 1232s]                 f"In the future `np.{attr}` will be defined as the "
[ 1232s]                 "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
[ 1232s]     
[ 1232s]         if attr in __former_attrs__:
[ 1232s]             raise AttributeError(__former_attrs__[attr])
[ 1232s]     
[ 1232s]         if attr in __expired_attributes__:
[ 1232s]             raise AttributeError(
[ 1232s]                 f"`np.{attr}` was removed in the NumPy 2.0 release. "
[ 1232s]                 f"{__expired_attributes__[attr]}"
[ 1232s]             )
[ 1232s]     
[ 1232s]         if attr == "chararray":
[ 1232s]             warnings.warn(
[ 1232s]                 "`np.chararray` is deprecated and will be removed from "
[ 1232s]                 "the main namespace in the future. Use an array with a string "
[ 1232s]                 "or bytes dtype instead.", DeprecationWarning, stacklevel=2)
[ 1232s]             import numpy.char as char
[ 1232s]             return char.chararray
[ 1232s]     
[ 1232s] >       raise AttributeError("module {!r} has no attribute "
[ 1232s]                              "{!r}".format(__name__, attr))
[ 1232s] E       AttributeError: module 'numpy' has no attribute 'alltrue'
[ 1232s] 
[ 1232s] /usr/lib64/python3.10/site-packages/numpy/__init__.py:410: AttributeError
```

And so on. Please fix before the 4.2.2 release

**Comment by radoering** [comment]
> The unit tests have been missed by this PR

Interesting. I assume ruff missed that because numpy is not imported directly in the unit test but via a star import from `wx.lib.floatcanvas.Utilities.BBox`. Lessons learned: ruff cannot replace a full-text search.

**Pull request merged**